packages/dbml-core/src/parse/dbml/src/lib/parser/nodes.ts

```diff
@@ -0,0 +1,575 @@
+
+  bodyOpenColon?: SyntaxToken;
+
+  bodyOpenBrace?: SyntaxToken;
+
+  body: ExpressionNode | BlockExpressionNode;
+
+  bodyCloseBrace?: SyntaxToken;
+
+  constructor({
+    type,
+    name,
+    as,
+    alias,
+    attributeList,
+    bodyOpenColon,
+    body,
+  }: {
+    type: SyntaxToken;
+    name?: NormalFormExpressionNode;
+    as?: SyntaxToken;
+    alias?: NormalFormExpressionNode;
+    attributeList?: ListExpressionNode;
+    bodyOpenColon?: SyntaxToken;
+    body: BlockExpressionNode | ExpressionNode;
+  }) {
+    this.start = type.offset;
+    this.end = body.end;
+    this.type = type;
+    this.name = name;
+    this.as = as;
+    this.alias = alias;
+    this.attributeList = attributeList;
+    this.bodyOpenColon = bodyOpenColon;
+    this.body = body;
+  }
+}
+
+export class FieldDeclarationNode implements SyntaxNode {
+  kind: SyntaxNodeKind.FIELD_DECLARATION = SyntaxNodeKind.FIELD_DECLARATION;
+
+  start: Readonly<number>;
+
+  end: Readonly<number>;
+
+  name: SyntaxToken;
+
+  valueOpenColon: SyntaxToken;
+
+  value: NormalFormExpressionNode;
+
+  attributeList?: ListExpressionNode;
+
+  constructor({
+    name,
+    valueOpenColon,
+    value,
+    attributeList,
+  }: {
+    name: SyntaxToken;
+    valueOpenColon: SyntaxToken;
+    value: NormalFormExpressionNode;
+    attributeList?: ListExpressionNode;
+  }) {
+    this.start = name.offset;
+    if (attributeList) {
+      this.end = attributeList.end;
+    } else {
+      this.end = value.end;
+    }
+    this.name = name;
+    this.valueOpenColon = valueOpenColon;
+    this.value = value;
+    this.attributeList = attributeList;
+  }
+}
+
+export class AttributeNode implements SyntaxNode {
+  kind: SyntaxNodeKind.ATTRIBUTE = SyntaxNodeKind.ATTRIBUTE;
+
+  start: Readonly<number>;
+
+  end: Readonly<number>;
+
+  name: SyntaxToken[];
+
+  valueOpenColon?: SyntaxToken;
+
+  value?: NormalFormExpressionNode;
+
+  constructor({
+    name,
+    valueOpenColon,
+    value,
+  }: {
+    name: SyntaxToken[];
+    valueOpenColon?: SyntaxToken;
+    value?: NormalFormExpressionNode;
+  }) {
+    if (name.length === 0) {
+      this.start = -1;
+      this.end = -1;
+    } else {
+      this.start = name[0].offset;
+      if (value) {
+        this.end = value.end;
+      } else {
+        const lastName = name[name.length - 1];
+        this.end = lastName.offset + lastName.length - 1;
+      }
+    }
+    this.name = name;
+    this.valueOpenColon = valueOpenColon;
+    this.value = value;
+  }
+}
+
+export type NormalFormExpressionNode =
+  | PrefixExpressionNode
+  | InfixExpressionNode
+  | PostfixExpressionNode
+  | LiteralElementExpressionNode
+  | FunctionApplicationNode
+  | BlockExpressionNode
+  | ListExpressionNode
+  | TupleExpressionNode
+  | CallExpressionNode
+  | PrimaryExpressionNode
+  | FunctionExpressionNode
+  | AccessExpressionNode;
+
+export type ExpressionNode =
+  | LiteralElementExpressionNode
+  | NormalFormExpressionNode
```

Comment:
This `NormalFormExpressionNode` also contains `LiteralElementExpressionNode` and `FunctionApplicationNode`, so what is the point of creating a new type `ExpressionNode`? :pepe-facepalm